src/python.cpp

```diff
@@ -992,8 +1115,120 @@ PyObject* get_np_nditer () {
   return np_nditer;
 }
 
+
+SEXP py_callable_as_function(SEXP callable, bool convert) {
+  SEXP f = PROTECT(eval_call(r_func_py_callable_as_function, callable));
+
+  // copy over class attribute
+  Rf_setAttrib(f, R_ClassSymbol, Rf_getAttrib(callable, R_ClassSymbol));
+
+  // save reference to underlying py_object
+  Rf_setAttrib(f, sym_py_object, callable);
+
+  UNPROTECT(1);
+  return f;
+}
+
+
+
+SEXP py_to_r_wrapper(SEXP x) {
+  SEXP x2 = eval_call_in_userenv(r_func_py_to_r_wrapper, x);
+  if(x == x2) // no method, py_to_r_wrapper.default() reflects
+    return(x);
+
+  // copy over all attributes ("class" and "py_object", typically)
+  PROTECT(x2);
+  SEXP a = ATTRIB(x);
+  while (CAR(a) != R_NilValue) {
```

Comment:
I think this wants to check `a != R_NilValue`; check how R iterates over the attributes pairlist in https://github.com/wch/r-source/blob/25ed2f1ac6256ece6b2f355f158c21196127acb5/src/main/attrib.c#L152-L162

Reply:
It's also worth checking if `Rf_copyMostAttrib()` does what you want, e.g. https://github.com/wch/r-source/blob/25ed2f1ac6256ece6b2f355f158c21196127acb5/src/main/attrib.c#L280-L299 -- that is, do you want to copy names / dimensions too?

Reply:
Good catch! I opted to not use `Rf_copyMostAttrib()` for backcompat. The previous R implementaiton did this (which would copy `dimnames` `dim` `names` too):
```r
attributes(x2) <- attributes(x)
```